src/mbgl/layout/symbol_instance.hpp

```diff
@@ -90,43 +136,149 @@ class SymbolInstance {
     const std::optional<SymbolQuads>& verticalIconQuads() const;
     void releaseSharedData();
 
+#if MLN_SYMBOL_GUARDS
+    /// Check all guard blocks
+    bool check(const std::source_location&) const;
+    /// Check that an index is in the valid range
+    bool checkIndex(const std::optional<std::size_t>& index, std::size_t size, const std::source_location&) const;
+    /// Check all indexes
+    bool checkIndexes(std::size_t textCount,
+                      std::size_t iconSize,
+                      std::size_t sdfSize,
+                      const std::source_location&) const;
+    /// Mark this item as failed (due to some external check) so that it cannot be used later
+    void forceFail() const;
+#else
+    bool check(std::string_view = {}) const { return true; }
+    bool checkIndex(const std::optional<std::size_t>&, std::size_t, std::string_view = {}) const { return true; }
+    bool checkIndexes(std::size_t, std::size_t, std::size_t, std::string_view = {}) const { return true; }
+    void forceFail() const {}
+#endif
+
+    const Anchor& getAnchor() const { return anchor; }
```

Comment:
Why are the getters needed now?

Reply:
Can you remove the ones that are not used?

None of the getters that return references seem to be used.

Reply:
They're used in `symbol_layout.cpp`, see line 855.  I think they're all used somewhere, I removed the one field that wasn't referenced anywhere.